Sources/FirebaseFirestore/QuerySnapshot+Swift.swift

```diff
@@ -6,3 +6,37 @@ import firebase
 import CxxShim
 
 public typealias QuerySnapshot = firebase.firestore.QuerySnapshot
+
+extension QuerySnapshot {
+  public var query: Query {
+    swift_firebase.swift_cxx_shims.firebase.firestore.query_snapshot_query(self)
+  }
+
+  public var metadata: SnapshotMetadata {
+    swift_firebase.swift_cxx_shims.firebase.firestore.query_snapshot_metadata(self)
+  }
+
+  public var isEmpty: Bool {
+    empty()
+  }
+
+  public var count: Int {
+    swift_firebase.swift_cxx_shims.firebase.firestore.query_snapshot_size(self)
+  }
+
+  public var documents: [DocumentSnapshot] {
+    Array<DocumentSnapshot>(swift_firebase.swift_cxx_shims.firebase.firestore.query_snapshot_documents(self))
+  }
+
+  public var documentChanges: [DocumentChange] {
+    Array<DocumentChange>(
+      swift_firebase.swift_cxx_shims.firebase.firestore.query_snapshot_document_changes(self, .exclude)
+    )
+  }
+
+  public func documentChanges(includeMetadataChanges: Bool) -> [DocumentChange] {
+    Array<DocumentChange>(
+      swift_firebase.swift_cxx_shims.firebase.firestore.query_snapshot_document_changes(self, includeMetadataChanges ? .include : .exclude)
```

Comment:
It looks like `query_snapshot_document_changes` includes a default value of `.exclude` in the C++ definition within `FirebaseFirestore.hh` should we reflect that default in this function definition as well? Probably not totally needed, but wanted to ask.

Reply:
No, the strategy I took is to declare a computed property `documentChanges` to align with the way the Obj C API is reflected to Swift.